docs/deployment/config_descriptions.md

```diff
@@ -4,51 +4,93 @@ title: Configuration Descriptions
-
-**The cumulus stack name must start with `<prefix>`** [^2]
-
-## buckets
-
-The buckets created in the [Create S3 Buckets](#create-s3-buckets) step. Buckets are defined in the config.yml with a key, name, and type. Types should be one of: internal, public, private, or protected. Multiple buckets of each type can be configured. A key is used for the buckets to allow for swapping out the bucket names easily.
-
-## useNgapPermissionBoundary
+# Cumulus Configuration
+
+## Overview
+
+### config.yml Explained
+
+| field | default     | description
+| ----- | ----------- | -----------
+| prefix | (required) | the name used as a prefix in all aws resources
+| prefixNoDash | (required) | prefix with no dash
+| urs_url | uat.urs | urs url used for OAuth
+| api_backend_url | apigateway backend url | the API backend url
+| api_distribution_url | apigateway dist url | the API url used for file distribution
+| shared_data_bucket | cumulus-data-shared | the bucket has the shared data artifacts
+| system_bucket | (required) | the bucket used for storing deployment artifacts
+| buckets | N/A | Configuration of buckets with key, bucket name, and type (i.e. internal, public private)
+| cmr.username | devseed | the username used for posting metadata to CMR
+| cmr.provider | CUMULUS | the provider used for posting metadata to CMR
+| cmr.clientId | CUMULUS | the clientId used to authenticate with the CMR
+| cmr.password | (required) | the password used to authenticate with the CMR
+| ems.provider | CUMULUS | the provider used for sending reports to EMS
+| vpc.vpcId | (required if ecs is used) | the vpcId used with the deployment
+| vpc.subnets | (required) | the subnets used
+| vpc.cidrIp | (required) | the IPv4 CIDR of the VPC
+| ecs.amiid | ami-9eb4b1e5 | amiid of an optimized ecs instance (differnet for each region)
+| ecs.instanceType | (required) | the instance type of the ec2 machine used for running ecs tasks
+| ecs.volumeSize | 50 | the storage on ec2 instance running the ecs tasks
+| ecs.availabilityZone | us-east-1a | the availibity zone used for launching ec2 machines
+| ecs.minInstances | 1 | min number of ec2 instances to launch in an autoscaling group
+| ecs.desiredInstances | 1 | desired number of ec2 instances needed in an autoscaling group
+| ecs.maxInstances | 2 | max number of ec2 instances to launch in an autoscaling group
+| es.name | es5 | name of the elasticsearch cluster
+| es.elasticSearchMapping | 4 | version number of the elasticsearch mapping used
+| es.version | 5.3 | elasticsearch software version
+| es.instanceCount | 1 | number of elasticsearch nodes
+| es.instanceType | t2.small.elasticsearch | size of the ec2 instance used for the elasticsearch
+| es.volumeSize | 35 | the storage used in each elasticsearch node
+| sns.\<name\> | N/A | name of the sns topic
+| sns.\<name\>.subscriptions.lambda.endpoint | sns2elasticsearch | lambda function triggered for each message in the topic
```

Comment:
This is pretty confusing and kinda misleading. The lambda function subscription for `sns.sftracker.subscriptions.lambda.endpoint` is `sns2elasticsearch`, but that value isn't the default for any SNS subscription you set up.

I'm not sure how best to document this because the endpoint value is an object using intrinsic references to a lambda function in the CF template. It's not just the name of a lambda function. Maybe this is the best we can do and the rest has to handled by pointing to the deployment package?

`| sns.\<name\>.subscriptions.\<subscription-name\>.endpoint |  | lambda function triggered for each message in the topic`

Reply:
We probably could refactor so that the config just references a lambda function name and our CF template uses the intrinsic functions to lookup the lambda. But that's probably out of scope for this work and I'm not sure if it's a better idea or not. Though I don't love users managing intrinsic functions in their config